### ISSUE TITLE
Object level masks

### ISSUE THREAD

**Issue by mariiak2021** [HarborYuan/ovsam#21]
Hi @HarborYuan,

Thank you for the great work! You mentioned in the paper that you are concentrating only object level masks using segment anything model. 

- What was done specifically to get object level masks only and avoid part masks? 
- What size of the point grid was used to train the model? Was it different from 32x32 original SAM grid?
- Where can I find the list of all classes list? is it possible to restrict the amount of classes to what I need?

Thank you!

**Comment by HarborYuan** [comment]
Hi @mariiak2021 ,

Thanks for your interest in our work.

1. The granularity of masks is mainly derived from the training data. You can specify your training data for training different forms of mask.
2. Our method does not have the concept of a grid. During training, we sample prompts from ground truth for training.
3. The class list is read from the dataset metainfo ([here](https://github.com/HarborYuan/ovsam/blob/126ba2b6422f5bec8bb9fad271c4cda16efde5ee/seg/datasets/coco_ins_ov.py#L52)). The open-vocabulary classifier needs an embedding generated by [`gen_cls`](https://github.com/HarborYuan/ovsam/blob/main/tools/gen_cls.py), which will read the class names in the metainfo to generate text embeddings.

Please let me know if you have any other questions.
